core-client/src/lib.rs

```diff
@@ -1465,26 +1464,35 @@ async fn do_custodian_recovery_init(
 
 async fn do_custodian_backup_recovery(
     core_endpoints: &HashMap<u32, CoreServiceEndpointClient<Channel>>,
+    sim_conf: &CoreClientConfig,
     custodian_context_id: RequestId,
     custodian_recovery_outputs: Vec<InternalCustodianRecoveryOutput>,
 ) -> anyhow::Result<()> {
+    // fetch the public keys of operators
+    // order of the verf keys should match the order of the core endpoints
+    let verf_keys = fetch_kms_verification_keys(sim_conf).await?;
+
     let mut req_tasks = JoinSet::new();
-    for (core_idx, ce) in core_endpoints.iter() {
+    for (endpoint_idx, ce) in core_endpoints.iter() {
```

Comment:
I am not sure this double loop is the best approach to handle this. Would it not be less fragile to instead have the `core_endpoint` be a hashmap of the verification key to the `CoreServiceEndpointClient`? Then it is possible to look up the end point directly for the key in a `custodian_recovery_outputs`

Reply:
yes I think it would be better if we remove the party ID from the core-client code completely since going forward there will be no more unique, long-term praty IDs. but I think that will need to be in another PR as that's used everywhere, in all the other core-client functions at the moment